src/core/reionization.c

```diff
@@ -1664,6 +1665,26 @@ void save_reion_input_grids(int snapshot)
   // fftw padded grids
   float* grid = (float*)calloc((size_t)local_nix * (size_t)ReionGridDim * (size_t)ReionGridDim, sizeof(float));
 
+  if (run_globals.params.Flag_IncludeSpinTemp) {
+    for (int ii = 0; ii < local_nix; ii++)
+      for (int jj = 0; jj < ReionGridDim; jj++)
+        for (int kk = 0; kk < ReionGridDim; kk++)
+          grid[grid_index(ii, jj, kk, ReionGridDim, INDEX_REAL)] =
+            (float)((grids->sfr)[grid_index(ii, jj, kk, ReionGridDim, INDEX_PADDED)] * UnitMass_in_g / UnitTime_in_s *
+                    SEC_PER_YEAR / SOLAR_MASS);
+    write_grid_float("sfr", grid, file_id, fspace_id, memspace_id, dcpl_id);
+#if USE_MINI_HALOS
+    for (int ii = 0; ii < local_nix; ii++)
+      for (int jj = 0; jj < ReionGridDim; jj++)
+        for (int kk = 0; kk < ReionGridDim; kk++)
+          grid[grid_index(ii, jj, kk, ReionGridDim, INDEX_REAL)] =
+            (float)((grids->sfrIII)[grid_index(ii, jj, kk, ReionGridDim, INDEX_PADDED)] * UnitMass_in_g /
+                    UnitTime_in_s * SEC_PER_YEAR / SOLAR_MASS);
+    write_grid_float("sfrIII", grid, file_id, fspace_id, memspace_id, dcpl_id);
+#endif
+  }
+
+  if (!sfr_only){
```

Comment:
I am not really understanding why you added this extra flag...

Reply:
cause in many cases, we don't want to save these grids at all snapshots. But in order to not load these higher-z sfr grids in RAM, we have to save them onto disc and load them again when needed.